src/main/scala/com/fulcrumgenomics/commons/collection/BetterBufferedIterator.scala

```diff
@@ -26,11 +26,17 @@ package com.fulcrumgenomics.commons.collection
 
 import com.fulcrumgenomics.commons.CommonsDef._
 
+/** A little trait to allow BetterBufferedIterator to override the headOption method. The latter method
+  * is not found in [[BetterBufferedIterator]] in 2.11.11, but in 2.12.2! */
+sealed trait HeadOption[A] {
```

Comment:
I don't think that this trait is needed at all since either clients will know the type to be BetterBufferedIterator and so `headOption` will exist, or it won't know the type beyond "BufferedIterator" in which case it can't call headOption.  I don't think adding the trait gives you anything beyond allowing you to add `override` below.